docs/deploy.md

```diff
@@ -236,3 +237,281 @@ The target mainchain address of a backward transfer has to be a valid address: s
+await payable.pay({ value: ethers.utils.parseEther("1") });
+console.log((await ethers.provider.getBalance(payable.address)).toString()); };
+return payable.address;
+};
+```
+Run the command to deploy the contract:
+
+```
+npx hardhat run ./scripts/Payable.ts
+```
+
+You will see the following execution results:
+
+```
+0xCE3Ab1fC99a77C4648c9c471c16d4D5e4568cE9A 1000000000000000000
+```
+
+
+## Interact with a Contract Instance
+
+To interact with your deployed smart contract, create a folder in your Hardhat project and name it, **tests/**. You will add test scripts that are specific to your contract instance as well as other contracts.
+
+### Interact with TestPayable: With ABI and Without ABI
+Interact with TestPayable by passing the ABI and without passing the ABI. Copy the script (below) into your **test/** folder:
+
+```
+const contractInteraction = async () => {
+const payableAddress = await deployPayable();
+// passing the ABI
+const payable = await ethers.getContractAt("TestPayable", payableAddress); await payable.pay({ value: ethers.utils.parseEther("1") });
+// without passing the ABI
+const [sender] = await ethers.getSigners(); await sender.sendTransaction({
+to: payableAddress,
+value: ethers.utils.parseEther("1"), });
+console.log((await ethers.provider.getBalance(payableAddress)).toString()); };
+contractInteraction();
+```
+
+Run the command to interact with this contract instance:
+
+```
+npx hardhat test
+```
+
+You will see the following execution results:
+
+```
+2000000000000000000
+```
+
+
+### Interact with TestPayable: Read → Call 
+Interact with TestPayable by using Read. Copy the script (below) into your **test/** folder:
+
+```
+const read = async () => {
+const payable = await ethers.getContractAt(
+"TestPayable",
+await deployPayable() );
+console.log(await payable.sentence()); };
+read();
+```
+
+Run the command to interact with this contract instance:
+
+```
+npx hardhat test
+```
+You will see the following execution results:
+
+```
+This is payable
+```
+
+### Interact with TestPayable: Call Non-Payable Function
+Interact with TestPayable by calling a non-payable function along with ZEN to verify that an error is thrown (as expected). Copy the script (below) into your **test/** folder:
+
+```
+const invalidPayment = async () => {
+const payable = await ethers.getContractAt(
+"TestPayable",
+await deployPayable() );
+// @ts-ignore
+await payable.dontPay({ value: ethers.utils.parseEther("1") }); };
+invalidPayment();
+```
+
+Run the command to interact with this contract instance:
+
+```
+npx hardhat test
+```
+
+You will see the following execution results:
+
+```
+/home/marko/Documents/Projects/hardhat-investigation/node_modules/@ethersproject/logg er/src.ts/index.ts:269
+const error: any = new Error(message); ^
+Error: non-payable method cannot override value (operation="overrides.value", value={"type":"BigNumber","hex":"0x0de0b6b3a7640000"}, code=UNSUPPORTED_OPERATION, version=contracts 5.7.0)
```

Comment:
Can you double check that this the expected result? Looks like there was an error

Reply:
Yes, I had Chien check for correctness and he verified that it is good to go.